javatests/arcs/sdk/HandleHolderBaseTest.kt

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.sdk
+
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import kotlin.test.assertFailsWith
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(JUnit4::class)
+class HandleHolderBaseTest {
+
+  @Test
+  fun handles_accessMissingHandle_throwsNoElemException() {
```

Comment:
I'd argue that this test isn't needed. You're testing a structure that's created solely for the purpose of allowing this test to pass.

The GeneratedHandleHolder test already covers verifying that we behave correctly for the HandleHolder emitted from codegen.

Reply:
You're right that this scenario is not likely to happen due to our generation, and it's covered by integration tests. However, isn't there value in defensive programming? And further, shouldn't we exercise our defenses, to ensure that the redundant safeguards are there? 

These checks are publicly accessible control flow that a user can exercise if they decided to hold the tool very wrong.

Reply:
I think I didn't quite make my argument correctly here: it's that the only reason this test needs to exist is that you've created a structure inside of this test that behaves in a way that this test validates. 

There's nothing inherent to HandleHolderBase that implies that we should create properties and implement them by delegating to a map. That's an implementation detail of the code generation utility, and so we should include tests coincident with code actually generated by the utility.

Reply:
Marking resolved: Removed these tests.